observations/util.py

```diff
@@ -3,15 +3,527 @@
+  print("Supports File Size:   [{}] ".format('Y'
+                                             if cap_dict['supports_file_size']
+                                             else 'N'))
+  print("File Size:            [{}] ".format(cap_dict['file_size']))
+  print("Additional GET params [{}]".format(cap_dict['addnl_params']))
+
+
+def check_capabilities(url, start=120, num_bytes=300, user_agent=None,
+                       _print=True):
+  """ Check capabilities of a specified URL. This function queries the URL
+  to get a response on the remote server capabilities.Requests for num_bytes
+  bytes of data to check total_file_size, range support, google drive specific
+  params
+
+  Args:
+    url: str.
+      URL string to the file.
+    start: int, optional.
+      Start position of the bytes of the file requested.
+    num_bytes: int, optional.
+      Number of bytes from the start byte requested.
+    user_agent: str, optional.
+      User agent which is sometimes required to query some domains for proper
+      download.
+    _print: bool, optional.
+       Print the remote server capabilities.
+
+  Returns
+    dict: Dictionary of remote server capabilities, required prior to full
+          download.
+  """
+  supports_range = False
+  supports_file_size = False
+  user_agent_default = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; rv:32.0' \
+                       ') Gecko/20100101 Firefox/32.0'
+  user_agent = user_agent_default if user_agent is None else user_agent
+  params = {}
+  session = requests.Session()
+  if 'drive.google.com' in url or 'docs.google.com' in url:
+    params = _gdrive_params(url, session)
+  range_header = {'User-Agent': user_agent,
+                  'Range': 'bytes=%d-%d' % (start, start + num_bytes)}
+  response = session.get(url, headers=range_header, params=params, stream=True)
+  content_type = response.headers.get('content-type')
+  try:
+    num_bytes_in_range = int(response.headers.get('content-length'))
+    if num_bytes_in_range == num_bytes + 1:
+      supports_range = True
+    else:
+      supports_range = False
+  except TypeError:
+    supports_range = False
+  try:
+    file_size = int(response.headers.get('content-range').split('/')[1])
+    supports_file_size = True
+  except TypeError:
+    file_size = 0
+    supports_file_size = False
+  capabilities_dict = {'content_type': content_type,
+                       'supports_resume': supports_range,
+                       'supports_file_size': supports_file_size,
+                       'file_size': file_size,
+                       'addnl_params': params,
+                       'user_agent': user_agent,
+                       'session': session
+                       }
+  if _print is True:
+    _print_capabilities(capabilities_dict)
+  return capabilities_dict
+
+
+def humanize_size(n):
+  """Convert file size in bytes to a friendly human readable form.
+  E.g humanize_size(1024) - '1 KB'.
```

Comment:
Examples are written at end of docstring like:
```
#### Examples

```python
humanize_size(1024)
## '1 KB'
```

See, e.g., https://github.com/blei-lab/edward/blob/master/edward/inferences/inference.py.